controllers/controllers/workloads/cforg_controller.go

```diff
@@ -23,43 +23,61 @@ import (
 
 	korifiv1alpha1 "code.cloudfoundry.org/korifi/controllers/api/v1alpha1"
 	"github.com/go-logr/logr"
-	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/hierarchical-namespaces/api/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
 // CFOrgReconciler reconciles a CFOrg object
 type CFOrgReconciler struct {
-	client client.Client
-	scheme *runtime.Scheme
-	log    logr.Logger
+	client                    client.Client
+	scheme                    *runtime.Scheme
+	log                       logr.Logger
+	packageRegistrySecretName string
 }
 
-func NewCFOrgReconciler(client client.Client, scheme *runtime.Scheme, log logr.Logger) *CFOrgReconciler {
+func NewCFOrgReconciler(client client.Client, scheme *runtime.Scheme, log logr.Logger, packageRegistrySecretName string) *CFOrgReconciler {
 	return &CFOrgReconciler{
-		client: client,
-		scheme: scheme,
-		log:    log,
+		client:                    client,
+		scheme:                    scheme,
+		log:                       log,
+		packageRegistrySecretName: packageRegistrySecretName,
 	}
 }
 
 const (
-	StatusConditionReady  = "Ready"
-	OrgNameLabel          = "cloudfoundry.org/org-name"
-	hierarchyMetadataName = "hierarchy"
-	APIVersion            = "korifi.cloudfoundry.org/v1alpha1"
+	StatusConditionReady = "Ready"
+	orgFinalizerName     = "cfOrg.korifi.cloudfoundry.org"
 )
 
 //+kubebuilder:rbac:groups=korifi.cloudfoundry.org,resources=cforgs,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=korifi.cloudfoundry.org,resources=cforgs/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=korifi.cloudfoundry.org,resources=cforgs/finalizers,verbs=update
 
-//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch
-//+kubebuilder:rbac:groups=hnc.x-k8s.io,resources=subnamespaceanchors,verbs=list;create;delete;watch
-//+kubebuilder:rbac:groups=hnc.x-k8s.io,resources=hierarchyconfigurations,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="rbac.authorization.k8s.io",resources=roles,verbs=create;patch;delete;get;list;watch
+//+kubebuilder:rbac:groups="rbac.authorization.k8s.io",resources=rolebindings,verbs=create;patch;delete;get;list;watch
+
+//+kubebuilder:rbac:groups="",resources=events,verbs=create;update
```

Comment:
The org reconciler deals with namespaces, secrets, roles and rolebindings. Therefore putting kubebuilder rbac annotations for pods, statefulsets, events, etc. looks weird to me. I know that those permissions should be granted to the korifi service account, but I would rather have them in reconcilers that actually deal with such sort of objects